File: datasets_v2/finance/gnucash/compose/dialog_account_delete.kt
import androidx.compose.ui.tooling.preview.Preview
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.lazy.LazyColumn
import androidx.compose.foundation.lazy.items
import androidx.compose.material3.*
import androidx.compose.runtime.*
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.paint
import androidx.compose.ui.graphics.Color
import androidx.compose.foundation.Image
import androidx.compose.ui.res.painterResource
import androidx.compose.ui.layout.ContentScale
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.sp
import androidx.compose.ui.unit.dp

@Composable
fun dialog_account_delete() {
  Column(modifier = Modifier.fillMaxWidth().fillMaxHeight()) {
    Column(modifier = Modifier.fillMaxWidth().wrapContentHeight().padding(10.0.dp)) {
      Text(text = "Transactions", modifier = Modifier.wrapContentWidth().wrapContentHeight())
      Text(text = "This account contains transactions. \\nWhat would you like to do with these transactions", modifier = Modifier.fillMaxWidth().wrapContentHeight(), fontSize = 12.sp)
      Row(modifier = Modifier.fillMaxWidth().fillMaxHeight()) {
        /* TODO: bind selection state */
        RadioButton(selected = false, onClick = {})
        Text("Delete sub-accounts")
      }
      Row(modifier = Modifier.wrapContentWidth().wrapContentHeight()) {
        /* TODO: bind selection state */
        RadioButton(selected = false, onClick = {})
        Text("Move to:")
      }
      val spinnerOptions_target_accounts_spinner = listOf("Select", "Option 1", "Option 2", "Option 3")
      var spinnerExpanded_target_accounts_spinner by remember { mutableStateOf(false) }
      var spinnerSelected_target_accounts_spinner by remember { mutableStateOf(spinnerOptions_target_accounts_spinner.first()) }
      Box(modifier = Modifier.fillMaxWidth()) {
        OutlinedButton(onClick = { spinnerExpanded_target_accounts_spinner = true }, modifier = Modifier.fillMaxWidth()) {
          Text(text = spinnerSelected_target_accounts_spinner)
        }
        DropdownMenu(expanded = spinnerExpanded_target_accounts_spinner, onDismissRequest = { spinnerExpanded_target_accounts_spinner = false }) {
          spinnerOptions_target_accounts_spinner.forEach { option ->
            DropdownMenuItem(
              text = { Text(option) },
              onClick = {
                spinnerSelected_target_accounts_spinner = option
                spinnerExpanded_target_accounts_spinner = false
              }
            )
          }
        }
      }
    }
    Column(modifier = Modifier.fillMaxWidth().wrapContentHeight().padding(10.0.dp)) {
      Text(text = "Transactions", modifier = Modifier.wrapContentWidth().wrapContentHeight())
      Text(text = "This account contains transactions. \\nWhat would you like to do with these transactions", modifier = Modifier.fillMaxWidth().wrapContentHeight(), fontSize = 12.sp)
      Row(modifier = Modifier.fillMaxWidth().fillMaxHeight()) {
        /* TODO: bind selection state */
        RadioButton(selected = false, onClick = {})
        Text("Delete sub-accounts")
      }
      Row(modifier = Modifier.wrapContentWidth().wrapContentHeight()) {
        /* TODO: bind selection state */
        RadioButton(selected = false, onClick = {})
        Text("Move to:")
      }
      val spinnerOptions_target_accounts_spinner = listOf("Select", "Option 1", "Option 2", "Option 3")
      var spinnerExpanded_target_accounts_spinner by remember { mutableStateOf(false) }
      var spinnerSelected_target_accounts_spinner by remember { mutableStateOf(spinnerOptions_target_accounts_spinner.first()) }
      Box(modifier = Modifier.fillMaxWidth()) {
        OutlinedButton(onClick = { spinnerExpanded_target_accounts_spinner = true }, modifier = Modifier.fillMaxWidth()) {
          Text(text = spinnerSelected_target_accounts_spinner)
        }
        DropdownMenu(expanded = spinnerExpanded_target_accounts_spinner, onDismissRequest = { spinnerExpanded_target_accounts_spinner = false }) {
          spinnerOptions_target_accounts_spinner.forEach { option ->
            DropdownMenuItem(
              text = { Text(option) },
              onClick = {
                spinnerSelected_target_accounts_spinner = option
                spinnerExpanded_target_accounts_spinner = false
              }
            )
          }
        }
      }
    }
    Column(modifier = Modifier.fillMaxWidth().fillMaxHeight().padding(top = 5.0.dp)) {
      Spacer(modifier = Modifier.weight(1f))
      Row(modifier = Modifier.fillMaxWidth().wrapContentHeight()) {
        Button(onClick = {}, enabled = true, modifier = Modifier.wrapContentHeight().weight(1.0f)) {
          Text(text = "Cancel", color = Color(0xFFFFAB00))
        }
        Button(onClick = {}, enabled = true, modifier = Modifier.wrapContentHeight().weight(1.0f)) {
          Text(text = "Save", color = Color(0xFFFFAB00))
        }
      }
    }
  }
}


@Preview(showBackground = true)
@Composable
fun dialog_account_delete_preview() {
    MyApplicationTheme {
        dialog_account_delete()
    }
}

File: experiments/finance/gnucash/swiftui/Dialog_account_delete.swift
import SwiftUI

struct Dialog_account_delete: View {
  var body: some View {
    VStack(alignment: .leading, spacing: 0) {
      VStack(alignment: .leading, spacing: 8) {
        Text("Transactions")
        Text("This account contains transactions. \\nWhat would you like to do with these transactions").font(.system(size: 12)).frame(maxWidth: .infinity, alignment: .leading)
        /* TODO: RadioButton -> Toggle fallback */
        Toggle("Delete sub-accounts", isOn: .constant(false)).frame(maxWidth: .infinity, maxHeight: .infinity)
        /* TODO: RadioButton -> Toggle fallback */
        Toggle("Move to:", isOn: .constant(false))
        Picker("", selection: .constant(0)) {
          Text("Select").tag(0)
          Text("Option 1").tag(1)
          Text("Option 2").tag(2)
          Text("Option 3").tag(3)
        }
        .pickerStyle(.menu)
        .frame(maxWidth: .infinity)
      }
      .frame(maxWidth: .infinity)
      .padding(10.0)
      VStack(alignment: .leading, spacing: 8) {
        Text("Transactions")
        Text("This account contains transactions. \\nWhat would you like to do with these transactions").font(.system(size: 12)).frame(maxWidth: .infinity, alignment: .leading)
        /* TODO: RadioButton -> Toggle fallback */
        Toggle("Delete sub-accounts", isOn: .constant(false)).frame(maxWidth: .infinity, maxHeight: .infinity)
        /* TODO: RadioButton -> Toggle fallback */
        Toggle("Move to:", isOn: .constant(false))
        Picker("", selection: .constant(0)) {
          Text("Select").tag(0)
          Text("Option 1").tag(1)
          Text("Option 2").tag(2)
          Text("Option 3").tag(3)
        }
        .pickerStyle(.menu)
        .frame(maxWidth: .infinity)
      }
      .frame(maxWidth: .infinity)
      .padding(10.0)
      VStack(alignment: .leading, spacing: 0) {
        Spacer()
        HStack(alignment: .center, spacing: 0) {
          Button(action: { }) {
            Text("Cancel").foregroundColor(Color(red: 1, green: 0.6705882352941176, blue: 0, opacity: 1))
          }
          .disabled(false)
          .buttonStyle(.plain)
          .frame(maxWidth: .infinity)
          Button(action: { }) {
            Text("Save").foregroundColor(Color(red: 1, green: 0.6705882352941176, blue: 0, opacity: 1))
          }
          .disabled(false)
          .buttonStyle(.plain)
          .frame(maxWidth: .infinity)
        }
        .frame(maxWidth: .infinity)
      }
      .frame(maxWidth: .infinity, maxHeight: .infinity)
      .padding(.top, 5.0)
    }
    .frame(maxWidth: .infinity, maxHeight: .infinity)
  }
}

struct Dialog_account_delete_Previews: PreviewProvider {
  static var previews: some View {
    Dialog_account_delete()
  }
}
